app/data/tests/test_views.py

```diff
@@ -199,6 +210,34 @@ def test_created_by_public_client(self):
         public_response_data = json.loads(self.public_client.get(url).content)
         self.assertTrue(all('created_by' not in result for result in public_response_data['results']))
 
+    def test_unicode_in_filters_number_field(self):
```

Comment:
So it seemed to me that we care more about the request succeeding and writing a `str` to Redis than we do that `generate_query_sql` returns `unicode`.

I was initially testing that `_cache_tile_sql` was getting `unicode`, but that wouldn't have caught one of the two bugs that were present (we were re-encoding a string as a string in two places, the second of which is in `_cache_tile_sql`). The way I have it is more integration-y, but since we're already testing both the application layer and the database it seemed okay.

I think I like this test as is but it might be good to write an actual unit test for `generate_query_sql`.

Reply:
Yeah, I thought that something along those lines might be the case; I don't mind this test, and I think having an additional unit test on `generate_query_sql` would also be great.

Reply:
I think I'm just going to merge this as is because I've already spent quite a bit of time on this.